### ISSUE TITLE
Create a flexible "prevalence module" for netsim

### ISSUE THREAD

**Issue by AdrienLeGuillou** [statnet/EpiModel#444]
The idea is to allow the user to add "epi trackers" at runtime, without the need to modify the modules themselves.

This idea is also implemented [in this branch of EpiModelHIV-p](https://github.com/EpiModel/EpiModelHIV-p/blob/SFO_PrEP/R/mod.prevalence.R).

The user has to provide as list of function through the `param$epi_funs` parameters. One such function can look like this:

``` r
epi_linked_6month <- function(dat, at) {
  with(dat$attr, {
    cond <-  tx.init.time - diag.time <= 26
    pop <- diag.status == 1 

    sum(cond & pop, na.rm = TRUE) / sum(pop, na.rm = TRUE)
  })
}
```

The module can be simplify compared to EpiModelHIV, by using the new `set_epi` function like so:

``` r
prevalence.net <- function(dat, at) {
  epi_funs <- get_params(dat, epi_funs)

  for (fn in names(epi_funs)) {
    dat <- set_epi(dat, fn, at, epi_funs[[fn]](dat, at) )
  }
}
```

This method allows the use of functional programming magic to create function factories that simplify the creation of a lot of trackers. For instance, if I want the proportion of black people that had a viral load below 200 at any point in the past year I would add this tracker:

``` r
param$epi_funs <- c(
  param$epi_funs, 
  "sup_dx_this_year.B" = epi_sup_dx_this_year(1)
)
```

that use the following function:

``` r
epi_sup_dx_this_year <- function(r_ind) { # r_ind is the race(s) of interest
  # the output of the function is the actual tracker. (Hence the term "function factory") 
  function(dat, at) {
    with(dat$attr, {

      cond <- at - vl.last.supp <= 52
      pop <- (at - diag.time <= 52 + 52 & at - diag.time >= 52 &
                race %in% r_ind)

      sum(cond & pop, na.rm = TRUE) / sum(pop, na.rm = TRUE)
    })
  }
}
```

for the same tracker but without race distinction, I could then use

``` r
param$epi_funs <- c(
  param$epi_funs, 
 "sup_dx_this_year" = epi_sup_dx_this_year(1:3)
)
```

More example of these functions [can be looked at here](https://github.com/EpiModel/EpiModel_SFO/blob/reload/R/utils-prev_funs.R)

**Comment by AdrienLeGuillou** [comment]
I think the best way to tackle this issue would be to add a new `trackers.net` module  and not add the new code to the existing `prevalence.net` module. This way, the functionality could be easily added to custom projects, at the end of the module list.

@smjenness , what are your thoughts on this?

**Comment by smjenness** [comment]
Sorry it took me forever to respond to this. Yes, I think that's the right way to go, so this is only needed in some subset of Epimodel use cases in which extension is necessary, so a separate top-level module seems right. I would call this some like prevalence_extension or something like that instead of trackers to show that it builds on what is in prevalence.net

**Comment by AdrienLeGuillou** [comment]
closed by #530
